core-bundle/src/EventListener/CsrfTokenCookieSubscriber.php

```diff
@@ -71,6 +71,11 @@ public function onKernelResponse(ResponseEvent $event): void
         if ($this->requiresCsrf($request, $response)) {
             $this->setCookies($request, $response);
         } else {
+            // Do not delete CSRF token cookie, if response is not successful (#2252)
+            if (!$response->isSuccessful()) {
+                return;
+            }
+
```

Comment:
```suggestion
        } elseif ($response->isSuccessful()) {
            // Only delete CSRF token cookie, if response is successful (#2252)
```

Reply:
I am not really in favor of `elseif` - here and in general. A separate condition makes it more readable imho.

Reply:
No, it does not. 🙈 Even my IDE says "this condition can be merged into the parent one". 😄